### PR TITLE
Pn 8027 fe rifattorizzare il componente empty state

### PR DESCRIPTION
**Pull request by NicolaGiornetta**
## Short description
Refactored EmptyState component and its usage.

## List of changes proposed in this pull request
- Removed most of the props about empty message structure with children prop in EmptyState component. Now you can use Trans component to make linear and complex empty messages.
- Reworked all the components which use EmptyState
- Reworked translations
- Reworked tests

## How to test
There are no visual changes because it should be the same. However you can test the purpose of EmptyState
Where to test:
- Notification list PA, PF, PG. Add a filter which makes not result.
- Delegates and delegators PA, PF, PG. Remove delegates / delegators to make empty list.
- API Keys in PA. Remove all the API keys to make empty list.
- App Status PA, PF, PG. Make sure to not have any disservice.

### CODE REVIEW

File: packages/pn-personafisica-webapp/src/components/Deleghe/Delegates.tsx [hunk 1]
```diff
@@ -113,6 +113,19 @@ const Delegates = () => {
     dispatch(setDelegatesSorting(s));
   };
 
+  const LinkAddDelegate = ({ children }: { children?: ReactNode }) => (
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Deleghe/Delegates.tsx:116]: Same as before

File: packages/pn-personafisica-webapp/src/components/Deleghe/MobileDelegates.tsx [hunk 2]
```diff
@@ -107,6 +107,19 @@ const MobileDelegates = () => {
     setShowCodeModal({ open: false, name: '', code: '' });
   };
 
+  const LinkAddDelegate = ({ children }: { children?: ReactNode }) => (
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Deleghe/MobileDelegates.tsx:110]: Same as before

File: packages/pn-personafisica-webapp/src/components/Notifications/DesktopNotifications.tsx [hunk 3]
```diff
@@ -134,35 +135,36 @@ const DesktopNotifications = ({
...
-      filtersApplied || currentDelegator
-        ? undefined
-        : {
-            emptyMessage: t('empty-state.second-message'),
-          },
-  };
-
   const showFilters = notifications?.length > 0 || filtersApplied;
 
+  const LinkRemoveFilters = ({ children }: { children?: ReactNode }) => (
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Notifications/DesktopNotifications.tsx:142]: Same as before

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/DelegatesByCompany.tsx [hunk 4]
```diff
@@ -143,6 +144,19 @@ const DelegatesByCompany = () => {
     }
   };
 
+  const LinkAddDelegate = ({ children }: { children?: ReactNode }) => (
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/DelegatesByCompany.tsx:147]: Same as before

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/DelegationsOfTheCompany.tsx [hunk 5]
```diff
@@ -341,6 +342,19 @@ const DelegationsOfTheCompany = () => {
     getDelegatorsData();
   }, [filters]);
 
+  const LinkRemoveFilters = ({ children }: { children?: ReactNode }) => (
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/DelegationsOfTheCompany.tsx:345]: Same as before

File: packages/pn-personafisica-webapp/src/components/Notifications/DesktopNotifications.tsx [hunk 6]
```diff
@@ -192,7 +194,30 @@ const DesktopNotifications = ({
           testId="notificationsTable"
         />
       ) : (
-        <EmptyState {...EmptyStateProps} />
+        <EmptyState
+          sentimentIcon={filtersApplied ? KnownSentiment.DISSATISFIED : KnownSentiment.NONE}
+        >
+          {filtersApplied && (
+            <Trans
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Notifications/DesktopNotifications.tsx:201]: I'm not sure that all the previous condition in EmptyStateProps are now respected

File: packages/pn-personafisica-webapp/src/components/Notifications/MobileNotifications.tsx [hunk 7]
```diff
@@ -254,7 +255,30 @@ const MobileNotifications = ({ notifications, sort, onChangeSorting, currentDele
           }}
         />
       ) : (
-        <EmptyState {...EmptyStateProps} />
+        <EmptyState
+          sentimentIcon={filtersApplied ? KnownSentiment.DISSATISFIED : KnownSentiment.NONE}
+        >
+          {filtersApplied && (
+            <Trans
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Notifications/MobileNotifications.tsx:262]: Same as before

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]
Check also these things:
- In the page tests, where there is the rendering of the empty state, add the check on the action click
- With the rework of the EmptyState, maybe some localizations are been cancelled. Check this but for now DON'T remove them

**Comment by NicolaGiornetta** [comment]
Translation labels replacements and additions:

notifiche.json of PA

removed:
"empty-state.message"
"empty-state.secondary-message"
"empty-state.secondary-action"
"empty-state.filter-message"
"empty-state.filter-action"

replaced with:
"empty-state.filtered" where before it was the sum of 
	"empty-state.message" +
	"menu.add" (of common.json) +
	"empty-state.secondary-message" +
	"empty-state.secondary-action"

"empty-state.no-notifications" where before it was the sum of
	"empty-state.filter-message" +
	"empty-state.filter-action"

"aria-label-remove-filters"
"aria-label-api-keys"
"aria-label-create-notification"

---------------------------------------------------------------------------

notifiche.json of PF

removed:
"empty-state.first-message"
"empty-state.action"
"empty-state.second-message"
"empty-state.filter-message"
"empty-state.filter-action"

replaced with:
"empty-state.filtered" where before it was the sum of
	"empty-state.filter-message" +
	"empty-state.filter-action"

"empty-state.no-notifications" where before it was the sum of
	"empty-state.first-message" +
	"empty-state.action" +
	"empty-state.second-message"
	
"empty-state.aria-label-remove-filters"
"empty-state.aria-label-route-contacts"

---------------------------------------------------------------------------

notifiche.json of PG

removed:
"empty-state.message"
"empty-state.filter-message",
"empty-state.filter-action"

replaced with:
"empty-state.filtered" where before it was the sum of
	"empty-state.filter-message" +
	"empty-state.filter-action"
	
"empty-state.no-notifications" where before it was "empty-state.message"

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**